### PR TITLE
Add table aws_redshift_parameter_group. Closes #190

### LINKED ISSUE #190: Add table aws_redshift_parameter_group
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.37.31/service/redshift/redshiftiface

### PR DESCRIPTION
**Pull request by anisadas**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_redshift_parameter_group []

PRETEST: tests/aws_redshift_parameter_group

TEST: tests/aws_redshift_parameter_group
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_redshift_parameter_group.named_test_resource: Creating...
aws_redshift_parameter_group.named_test_resource: Creation complete after 4s [id=turbottest47859]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:redshift:us-east-1:986325076436:parametergroup:turbottest47859
resource_name = turbottest47859

Running SQL query: test-get-query.sql
[
  {
    "family": "redshift-1.0",
    "name": "turbottest47859",
    "tags": {
      "name": "turbottest47859"
    }
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "name": "turbottest47859",
    "tags": {
      "name": "turbottest47859"
    },
    "tags_src": [
      {
        "Key": "name",
        "Value": "turbottest47859"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "name": "turbottest47859",
    "partition": "aws",
    "region": "us-east-1",
    "tags": {
      "name": "turbottest47859"
    },
    "title": "turbottest47859"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:redshift:us-east-1:986325076436:parametergroup:turbottest47859"
    ],
    "name": "turbottest47859",
    "region": "us-east-1",
    "tags": {
      "name": "turbottest47859"
    },
    "title": "turbottest47859"
  }
]
✔ PASSED

POSTTEST: tests/aws_redshift_parameter_group

TEARDOWN: tests/aws_redshift_parameter_group

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```

select
  name,
  description,
  family
from
  aws_redshift_parameter_group;

+------------------------+------------------------------------------+--------------+
| name                   | description                              | family       |
+------------------------+------------------------------------------+--------------+
| turbottest30422        | Managed by Terraform                     | redshift-1.0 |
| turbottest1234         | Arghya will delete this                  | redshift-1.0 |
| default.redshift-1.0   | Default parameter group for redshift-1.0 | redshift-1.0 |
| turbottest49024        | Managed by Terraform                     | redshift-1.0 |
| default.redshift-1.0   | Default parameter group for redshift-1.0 | redshift-1.0 |
| turbottest25267        | Managed by Terraform                     | redshift-1.0 |
| turbot-turbottest27452 | Created by Turbot                        | redshift-1.0 |
+------------------------+------------------------------------------+--------------+

select
  name,
  p ->> 'ParameterName' as parameter_name,
  p ->> 'ParameterValue' as parameter_value,
  p ->> 'Description' as description,
  p ->> 'Source' as source,
  p ->> 'DataType' as data_type,
  p ->> 'ApplyType' as apply_type,
  p ->> 'IsModifiable' as is_modifiable,
  p ->> 'AllowedValues' as allowed_values,
  p ->> 'MinimumEngineVersion' as minimum_engine_version
from
  aws_redshift_parameter_group,
  jsonb_array_elements(parameters) as p
where
  p ->> 'ParameterName' = 'require_ssl'
  and p ->> 'ParameterValue' = 'false';

+------------------------+----------------+-----------------+-----------------------------------------+----------------+-----------+------------+---------------+----------------+-------------
| name                   | parameter_name | parameter_value | description                             | source         | data_type | apply_type | is_modifiable | allowed_values | minimum_engi
+------------------------+----------------+-----------------+-----------------------------------------+----------------+-----------+------------+---------------+----------------+-------------
| turbottest25267        | require_ssl    | false           | require ssl for all databaseconnections | user           | boolean   | static     | true          | true,false     | <null>      
| turbottest30422        | require_ssl    | false           | require ssl for all databaseconnections | engine-default | boolean   | static     | true          | true,false     | <null>      
| turbot-turbottest27452 | require_ssl    | false           | require ssl for all databaseconnections | user           | boolean   | static     | true          | true,false     | <null>      
| turbottest49024        | require_ssl    | false           | require ssl for all databaseconnections | engine-default | boolean   | static     | true          | true,false     | <null>      
| default.redshift-1.0   | require_ssl    | false           | require ssl for all databaseconnections | engine-default | boolean   | static     | true          | true,false     | <null>      
| default.redshift-1.0   | require_ssl    | false           | require ssl for all databaseconnections | engine-default | boolean   | static     | true          | true,false     | <null>      
+------------------------+----------------+-----------------+-----------------------------------------+----------------+-----------+------------+---------------+----------------+-------------


```
</details>

### CODE REVIEW

File: docs/tables/aws_redshift_parameter_group.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+select
+  name,
+  description,
+  family
+from
+  aws_redshift_parameter_group;
+```
+
+
+### Details of several parameters which are associated with each parameter group
```
Comment (Paulami30) [inline docs/tables/aws_redshift_parameter_group.md:18]:
```suggestion
### Details of the parameters associated with each parameter group
```

File: docs/tables/aws_redshift_parameter_group.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Table: aws_redshift_parameter_group
+
+A parameter group contains a WLM configuration and a set of cluster parameters. It optimizes query performance.
```
Comment (Paulami30) [inline docs/tables/aws_redshift_parameter_group.md:3]:
```suggestion
A parameter group is a group of parameters that apply to all of the databases that you create in the cluster. 

```

File: docs/tables/aws_redshift_parameter_group.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  aws_redshift_parameter_group;
+```
+
+
+### Details of several parameters which are associated with each parameter group
+
+```sql
+select
+  name,
+  jsonb_pretty(parameters)
```
Comment (Paulami30) [inline docs/tables/aws_redshift_parameter_group.md:23]: expand the json object

File: docs/tables/aws_redshift_parameter_group.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Table: aws_redshift_parameter_group
+
+A parameter group is a group of parameters that apply to all of the databases that you create in the cluster.
```
Comment (Paulami30) [inline docs/tables/aws_redshift_parameter_group.md:3]:
```suggestion
A parameter group is a group of parameters that apply to all of the databases that you create in the cluster. 

```
```suggestion
A parameter group is a group of parameters that apply to all of the databases that you create in the cluster. 
```

Comment (Paulami30) [inline docs/tables/aws_redshift_parameter_group.md:3]:
```suggestion
A parameter group is a group of parameters that apply to all of the databases that you create in the cluster. 

```

File: aws-test/tests/aws_redshift_parameter_group/test-get-query.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+select family, name , tags
```
Comment (Paulami30) [inline aws-test/tests/aws_redshift_parameter_group/test-get-query.sql:1]:
```suggestion
select family, name, tags
```

File: aws-test/tests/aws_redshift_parameter_group/test-hydrate-query.sql [hunk 6]
```diff
@@ -0,0 +1,3 @@
+select name , tags , tags_src
```
Comment (Paulami30) [inline aws-test/tests/aws_redshift_parameter_group/test-hydrate-query.sql:1]:
```suggestion
select name, tags, tags_src
```

File: aws-test/tests/aws_redshift_parameter_group/test-list-query.sql [hunk 7]
```diff
@@ -0,0 +1,3 @@
+select name , partition , region  , tags , title
```
Comment (Paulami30) [inline aws-test/tests/aws_redshift_parameter_group/test-list-query.sql:1]:
```suggestion
select name, partition, region , tags, title
```

Comment (Paulami30) [reply aws-test/tests/aws_redshift_parameter_group/test-list-query.sql:1]: Fix spaces in the sql files

File: docs/tables/aws_redshift_parameter_group.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+select
+  name,
+  description,
+  family
+from
+  aws_redshift_parameter_group;
+```
+
+
+### Details of the parameters associated with each parameter group
```
Comment (Paulami30) [inline docs/tables/aws_redshift_parameter_group.md:19]:
```suggestion
### Get the details of the parameters associated with each parameter group
```

File: aws/table_aws_redshift_parameter_group.go [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsRedshiftParameterGroups(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// TODO put me in helper function
```
Comment (Paulami30) [inline aws/table_aws_redshift_parameter_group.go:86]:
```suggestion

	// TODO put me in helper function
```

File: aws/table_aws_redshift_parameter_group.go [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+			ShouldIgnoreError: isNotFoundError([]string{"ClusterParameterGroupNotFound"}),
+			Hydrate:           getAwsRedshiftParameterGroup,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listAwsRedshiftParameterGroups,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "name",
```
Comment (Paulami30) [inline aws/table_aws_redshift_parameter_group.go:30]:
```suggestion
				Name:        "parameter_group_name",
```

File: aws/plugin.go [hunk 11]
```diff
@@ -87,6 +87,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_lambda_alias":                       tableAwsLambdaAlias(ctx),
 			"aws_lambda_function":                    tableAwsLambdaFunction(ctx),
 			"aws_lambda_version":                     tableAwsLambdaVersion(ctx),
+			"aws_redshift_parameter_group":           tableAwsRedshiftParameterGroup(ctx),
```
Comment (Paulami30) [inline aws/plugin.go:90]: sort the list

File: docs/tables/aws_redshift_parameter_group.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+  p ->> 'Description' as description,
+  p ->> 'ParameterName' as parameter_name,
+  p ->> 'ParameterValue' as parameter_value,
+  p ->> 'Source' as source,
+  p ->> 'DataType' as data_type,
+  p ->> 'ApplyType' as apply_type
+from
+  aws_redshift_parameter_group,
+  jsonb_array_elements(parameters) as p;
+```
```
Comment (Paulami30) [inline docs/tables/aws_redshift_parameter_group.md:33]: Rearrange this and add the other fields like AllowedValues, IsModifiable, MinimumEngineVersion
```
select
  name,
  p ->> 'ParameterName' as parameter_name,
  p ->> 'ParameterValue' as parameter_value,
  p ->> 'Description' as description,
  p ->> 'Source' as source,
  p ->> 'DataType' as data_type,
  p ->> 'ApplyType' as apply_type
from
  aws_redshift_parameter_group,
  jsonb_array_elements(parameters) as p;
```

**Review by Paulami30 — APPROVED** [review]

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
Pls check the comment.

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Comment by rajmohanty17** [comment]
Thanks for changing the doc.

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**